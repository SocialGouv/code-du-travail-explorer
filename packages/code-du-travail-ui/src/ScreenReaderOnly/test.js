import React from "react";
import { render } from "react-testing-library";
import ScreenReaderOnly from ".";

describe("<ScreenReaderOnly />", () => {
  test("should render", () => {
    const { container } = render(<ScreenReaderOnly />);
    expect(container).toMatchSnapshot();
  });
});
