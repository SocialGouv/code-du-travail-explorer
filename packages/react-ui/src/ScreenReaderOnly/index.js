import PropTypes from "prop-types";
import React from "react";
import styled from "styled-components";

export const ScreenReaderOnly = ({ type, ...props }) => {
  const Component = type === "inline" ? InlineSR : BlockSR;
  return <Component {...props} />;
};

ScreenReaderOnly.propTypes = {
  type: PropTypes.oneOf(["block", "inline"]),
};

ScreenReaderOnly.defaultProps = {
  type: "block",
};

const InlineSR = styled.span`
  &:not(:focus):not(:active) {
    position: absolute;
    width: 1px;
    height: 1px;
    margin: -1px;
    padding: 0;
    overflow: hidden;
    border: 0;
    clip: rect(0, 0, 0, 0);
  }
`;

const BlockSR = styled.div`
  &:not(:focus):not(:active) {
    position: absolute;
    width: 1px;
    height: 1px;
    margin: -1px;
    padding: 0;
    overflow: hidden;
    border: 0;
    clip: rect(0, 0, 0, 0);
  }
`;
