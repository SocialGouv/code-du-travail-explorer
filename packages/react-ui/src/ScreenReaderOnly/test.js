import { render } from "@testing-library/react";
import React from "react";

import { ScreenReaderOnly } from ".";

describe("<ScreenReaderOnly />", () => {
  test("should render", () => {
    const { container } = render(<ScreenReaderOnly>SR Only</ScreenReaderOnly>);
    expect(container).toMatchSnapshot();
  });
});
