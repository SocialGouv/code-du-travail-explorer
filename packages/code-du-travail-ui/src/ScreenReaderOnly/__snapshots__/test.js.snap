// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScreenReaderOnly /> should render 1`] = `
.c0:not(:focus):not(:active) {
  position: absolute;
  width: 1px;
  height: 1px;
  margin: -1px;
  padding: 0;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  border: 0;
}

<div>
  <div
    class="c0"
  />
</div>
`;
